### PR TITLE
feat: parse SharedArrayBuffer and validate AllowShared

### LINKED ISSUE #746: Give SharedArrayBuffer a dedicated type
See https://github.com/whatwg/webidl/pull/1311.

### PR DESCRIPTION
**Pull request by saschanaz**
This patch closes #746 and includes:
- [x] A relevant test
- [x] A relevant documentation update

### CODE REVIEW

File: lib/productions/type.js [hunk 1]
```diff
@@ -194,6 +194,25 @@ export class Type extends Base {
   *validate(defs) {
     yield* this.extAttrs.validate(defs);
 
+    if (this.idlType === "BufferSource") {
+      // XXX: For now this is a hack. Consider moving parents' extAttrs into types as the spec says:
+      // https://webidl.spec.whatwg.org/#idl-annotated-types
+      for (const extAttrs of [this.extAttrs, this.parent?.extAttrs]) {
```
Comment (marcoscaceres) [inline lib/productions/type.js:200]: Could this be `[this.extAttrs, this.parent?.extAttrs].flat()` to avoid the second for loop?

Comment (marcoscaceres) [reply lib/productions/type.js:200]: yeah, maybe also... to flatten to code a little more:
``` 
for (const extAttr of [this.extAttrs, this.parent?.extAttrs]) {
   if (extAttr.name !== "AllowShared") continue;
   // the rest... 
}
```

**Review by marcoscaceres — APPROVED** [review]
Just a style nit.

File: lib/productions/type.js (hunk 1 above)
Comment (saschanaz) [reply lib/productions/type.js:200]: That way the autofix won't know which array includes the attr and it'll need to rescan 😬

**Pull request merged**